core/sql/executor/ExExeUtilGet.cpp

```diff
@@ -5171,14 +5227,23 @@ short ExExeUtilHiveMDaccessTcb::work()
 	    str_pad(infoCol->dtQualifier, 28, ' ');
 
 	    if (infoCol->fsDatatype == REC_DATETIME)
-	      {
-		// hive currently only supports timestamp
+	    {
+              if(infoCol->colSize > 10) {
 		infoCol->dtCode = SQLDTCODE_TIMESTAMP;
 		infoCol->colScale = 6;
 		str_cpy(infoCol->dtQualifier, "(6)", 28, ' ');
 		infoCol->dtStartField = 1;
 		infoCol->dtEndField = 6;
-	      }
+              }
+              else
+              {
+		infoCol->dtCode = SQLDTCODE_DATE;
+		infoCol->colScale = 0;
+	        str_pad(infoCol->dtQualifier, 28, ' ');
+		infoCol->dtStartField = 1;
+		infoCol->dtEndField = 6;
```

Comment:
The dtEndField should be 3 I think. Please see common/DatetimeType.h. SqlDate goes from REC_DATE_YEAR to REC_DATE_DAY. I don't if this minor error will lead to a noticeable issue. If it could be addressed in a later PR that will be good.